manifest/fusing_manifest_provider.go

```diff
@@ -0,0 +1,109 @@
+package manifest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-f3/ec"
+	"go.uber.org/multierr"
+	"golang.org/x/sync/errgroup"
+)
+
+// HeadGetter is the minimal subset of github.com/filecoin-project/go-f3/ec.Backend required by the
+// MixedManifestProvider.
+type HeadGetter interface {
+	GetHead(context.Context) (ec.TipSet, error)
+}
```

Comment:
I tried to find a way to avoid this, but couldn't. I would need to have sent some kind of "ranked list" of possible manifests, letting the F3 module decide between them based on the current epoch, or something like that. It was looking pretty nasty.